### PR TITLE
fix(foundation): remove readonly from Slider

### PR DESCRIPTION
**Pull request by olaf-k**
## 📖 Description

This PR removes the `readonly` state for the Slider component in an effort to keep it close to the `input range`'s behavior.
[The specs for the latter can be found here](https://html.spec.whatwg.org/multipage/input.html#range-state-(type=range)) (see the 'Bookkeeping details' part after the examples).
image

See PR #6438 for a similar update to the radio button.

## 👩‍💻 Reviewer Notes

As it is a removal, the change is pretty straightforward. Code/tests/doc should be covered.
`slider-label` should not be affected as far as I can tell.

### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Comment by olaf-k** [comment]
@microsoft-github-policy-service agree company="Vonage"

File: change/@microsoft-fast-foundation-3285b619-2c39-4c8f-98c8-b839c81f85ea.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "removing readonly for Slider",
+  "packageName": "@microsoft/fast-foundation",
+  "email": "olaf-k@users.noreply.github.com",
+  "dependentChangeType": "patch"
```
Comment (nicholasrice) [inline change/@microsoft-fast-foundation-3285b619-2c39-4c8f-98c8-b839c81f85ea.json:6]:
```suggestion
  "dependentChangeType": "prerelease"
```

File: packages/web-components/fast-foundation/src/slider/slider.ts [hunk 2]
```diff
@@ -290,10 +275,6 @@ export class FASTSlider extends FormAssociatedSlider implements SliderConfigurat
     }
 
     protected keypressHandler = (e: KeyboardEvent) => {
-        if (this.readOnly) {
```
Comment (EisenbergEffect) [inline packages/web-components/fast-foundation/src/slider/slider.ts:293]: Should this instead be a check for `if (this.disabled)` like we have in the mouse event handlers?

**Review by EisenbergEffect — CHANGES_REQUESTED** [review]
This PR seems very similar in nature to the one we have for radio groups that makes the same set of changes. Like that one, I want to make sure we don't have a bug in the key handler. Shouldn't it be checking for disabled state like the mouse handler or does the platform handle that automatically?

File: packages/web-components/fast-foundation/src/slider/slider.ts (hunk 2 above)
Comment (olaf-k) [reply packages/web-components/fast-foundation/src/slider/slider.ts:293]: AFAICT, the disabled state prevents focus so keyboard events shouldn't register.

Comment (EisenbergEffect) [reply packages/web-components/fast-foundation/src/slider/slider.ts:293]: Great! I thought that might be the case but wasn't sure.

**Review by EisenbergEffect — APPROVED** [review]

**Comment by olaf-k** [comment]
hey team! do you think it'd be possible to push this forward to complete our great [ReadOnly](https://github.com/microsoft/fast/pull/6438) [Purge](https://github.com/microsoft/fast/pull/6515)? 😁

**Review by chrisdholt — APPROVED** [review]

**Review by bheston — APPROVED** [review]

**Pull request closed without merging**